cmd/kube-apiserver/app/server.go

```diff
@@ -569,7 +575,43 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 	}
 
 	if s.Etcd.EncryptionProviderConfigFilepath != "" {
-		transformerOverrides, err := encryptionconfig.GetTransformerOverrides(s.Etcd.EncryptionProviderConfigFilepath)
+		// Obtain GCE cloud for a Google Cloud KMS connection if needed.
+		// gce package cannot be accessed inside staging, which is why this function is passed from here.
+		getGCECloud := func() (*kms.GCECloud, error) {
+			cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
+			if err != nil {
+				return nil, err
+			}
+
+			var cloudkmsService *cloudkms.Service
+			var projectID string
+
+			// Safe when cloud is nil too.
+			if gke, ok := cloud.(*gce.GCECloud); ok {
+				// Hosting on GCE/GKE with Google KMS encryption provider
+				cloudkmsService = gke.GetKMSService()
+
+				// Project ID is assumed to be the user's project unless there
+				// is an override in the configuration file
+				projectID = gke.GetProjectID()
+			} else {
+				// Outside GCE/GKE. Requires GOOGLE_APPLICATION_CREDENTIALS environment variable.
```

Comment:
https://developers.google.com/identity/protocols/application-default-credentials

I will add this to the comment too, thanks for pointing out!